mne/preprocessing/eyetracking/calibration.py

```diff
@@ -0,0 +1,289 @@
+            f"  screen resolution: {screen_resolution} pixels\n"
+        )
+
+    def copy(self):
+        """Copy the instance.
+
+        Returns
+        -------
+        cal : instance of Calibration
+            The copied Calibration.
+        """
+        return deepcopy(self)
+
+    def __setitem__(self, key, value):
+        """Make sure that some keys are caste as numpy arrays.
+
+        Because methods like plot expect numpy arrays.
+        """
+        if key in ("positions", "offsets", "gaze") and isinstance(value, (tuple, list)):
+            logger.info("Converting %s to numpy array", key)
+            value = np.array(value)
+        super().__setitem__(key, value)
+
+    def plot(self, title=None, show_offsets=False, origin="top-left", show=True):
+        """Visualize calibration.
+
+        Parameters
+        ----------
+        title : str
+            The title to be displayed. Defaults to ``None``, which uses a generic title.
+        show_offsets : bool
+            Whether to display the offset (in visual degrees) of each calibration
+            point or not. Defaults to ``False``.
+        origin : str
+            What should be considered the origin of the screen. Can be ``'top-left'``,
+            ``'top-right'``, ``'bottom-left'``, or ``'bottom-right'``. Defaults to
+            ``'top-left'`` because for most monitors, pixel coordinate ``(0,0)``, often
+            referred to as origin, is at the top left of corner of the screen.
+        show : bool
+            Whether to show the figure or not. Defaults to ``True``.
+
+        Returns
+        -------
+        fig : instance of matplotlib.figure.Figure
+            The resulting figure object for the calibration plot.
+        """
+        import matplotlib.pyplot as plt
+
+        msg = "positions and gaze keys must both be 2D numpy arrays."
+        assert isinstance(self["positions"], np.ndarray), msg
+        assert isinstance(self["gaze"], np.ndarray), msg
+
+        fig, ax = plt.subplots(constrained_layout=True)
+        px, py = self["positions"].T
+        gaze_x, gaze_y = self["gaze"].T
+
+        if title is None:
+            ax.set_title(f"Calibration ({self['eye']} eye)")
+        else:
+            ax.set_title(title)
+        ax.set_xlabel("x (pixels)")
+        ax.set_ylabel("y (pixels)")
+
+        # Display avg_error and max_error in the top left corner
+        text = (
+            f"avg_error: {self['avg_error']} deg.\nmax_error: {self['max_error']} deg."
+        )
+        ax.text(
+            0,
+            1.01,
+            text,
+            transform=ax.transAxes,
+            verticalalignment="baseline",
+            fontsize=8,
+        )
+
+        msg = (
+            "origin must be 'top-left', 'top-right', 'bottom-left', or 'bottom-right."
+            f" got {origin}"
+        )
+        assert origin in ("top-left", "top-right", "bottom-left", "bottom-right"), msg
```

Comment:
`assert` should only be used in tests, or to assert something that we know should be true based on how we (internally) have coded something (which I think is the case for the `assert` statements farther up having to do with `ndarray` -- the `__setitem__` should ensure this for us). It should not be used for user input checks. This is because `assert` statements can be skipped by running `python` in an optimized mode. So instead you should use a helper like:
```suggestion
        _check_option('origin', origin, ("top-left", "top-right", "bottom-left", "bottom-right"))
```

Reply:
Oh, okay I see. I think I have a few asserts in this code that I should replace then...

is ``_check_ption`` an mne utility function?

Reply:
Yep, make friends with `git grep` and it will tell you :)
```
~/python/mne-python$ git grep _check_option
...
mne/baseline.py:from .utils import logger, verbose, _check_option, _validate_type
mne/baseline.py:        _check_option(
mne/beamformer/_compute_beamformer.py:    _check_option,
mne/beamformer/_compute_beamformer.py:    _check_option("pick_ori", pick_ori, ("normal", "max-power", "vector", None))
```
```

Reply:
hot tip!